dashboards-observability/common/utils/query_utils.ts

```diff
@@ -39,16 +41,19 @@ export const preprocessQuery = ({
   // convert to moment
   const start = datemath.parse(startTime)?.format(DATE_PICKER_FORMAT);
   const end = datemath.parse(endTime)?.format(DATE_PICKER_FORMAT);
-  const tokens = rawQuery
-    .replaceAll(PPL_NEWLINE_REGEX, "")
-    .match(PPL_INDEX_INSERT_POINT_REGEX);
-
+  const tokens = rawQuery.replaceAll(PPL_NEWLINE_REGEX, '').match(PPL_INDEX_INSERT_POINT_REGEX);
+  
   if (isEmpty(tokens)) return finalQuery;
-  finalQuery = `${tokens![1]}=${
-    tokens![2]
-  } | where ${timeField} >= '${start}' and ${timeField} <= '${end}'${
-    tokens![3]
-  }`;
 
-  return finalQuery;
-};
+  if (isLiveQuery){
+    finalQuery = `${tokens![1]}=${tokens![2]} | where ${timeField} >= '${start}' and ${timeField} <= '${end}'${tokens![3]} | sort - ${timeField}`;
+  } else {
+    finalQuery = `${tokens![1]}=${
+      tokens![2]
+    } | where ${timeField} >= '${start}' and ${timeField} <= '${end}'${
+      tokens![3]
+    }`;
+  };
+  return finalQuery; 
```

Comment:
i don't think you need to rewrite the part to insert timestamp filter?
```suggestion
  finalQuery = `${tokens![1]}=${
    tokens![2]
  } | where ${timeField} >= '${start}' and ${timeField} <= '${end}'${
    tokens![3]
  }`;
  if (isLiveQuery){
    finalQuery = finalQuery + ' | sort - ${timeField}`;
  }
  return finalQuery; 
```

Reply:
this works too!